proposals/intptr-operators.md

```diff
@@ -0,0 +1,54 @@
+# Operators should be exposed for `System.IntPtr` and `System.UIntPtr`
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+The CLR supports a set of operators for the `System.IntPtr` and `System.UIntPtr` types (`native int`). These operators can be seen in `III.1.5` of the Common Language Infrastructure specification (`ECMA-335`). However, these operators are not supported by C#.
+
+Language support should be provided for the full set of operators supported by `System.IntPtr` and `System.UIntPtr`. These operators are: `Add`, `Divide`, `Multiply`, `Remainder`, `Subtract`, `Negate`, `Equals`, `Compare`, `And`, `Not`, `Or`, `XOr`, `ShiftLeft`, `ShiftRight`.
+
+## Motivation
+[motivation]: #motivation
+
+Today, users can easily write C# applications targeting multiple platforms using various tools and frameworks, such as: `Xamarin`, `.NET Core`, `Mono`, etc...
+
+When writing cross-platform code, it is often necessary to write interop code that interacts with a particular target platform in a specific manner. This could include writing graphics code, calling some System API, or interacting with an existing native library.
+
+This interop code often has to deal with handles, unmanaged memory, or even just platform-specific sized integers.
+
+The runtime provides support for this by defining a set of operators that can be used on the `native int` (`System.IntPtr`) and `native unsigned int` (`System.UIntPtr`) primtive types.
+
+C# has never supported these operators and so users have to workaround the issue. This often increases code complexity and lowers code maintainability.
+
+As such, the language should begin to support these operators to help advance the language to better support these requirements.
+
+## Detailed design
+[design]: #detailed-design
+
+The full set of operators supported are defined in `III.1.5` of the Common Language Infrastructure specification (`ECMA-335`). The specification is available here: [https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf)
```

Comment:
Please describe (all of) the set of operators to be supported in a form consistent with the C# specification so we can better understand the implications for overload resolution, e.g.

``` c#
System.IntPtr operator +(System.IntPtr, System.IntPtr);
System.IntPtr operator +(System.Int32, System.IntPtr);
System.IntPtr operator +(System.IntPtr, System.Int32);
```

Is `0` a valid value that can be used where a value of type `System.IntPtr` is expected?